rfc/rfc-43/rfc-43.md

```diff
@@ -0,0 +1,316 @@
+
+2. Using Async compaction/clustering, in this mode the job execute async but also sharing the resource with HUDI to
+   write a job that may affect the stability of job writing, which is not what the user wants to see.
+
+3. Using independent compaction/clustering job is a better way to schedule the job, in this mode the job execute async
+   and do not sharing resources with writing job, but also has some questions:
+    1. Users have to enable lock service providers so that there is not data loss. Especially when compaction/clustering
+       is getting scheduled, no other writes should proceed concurrently and hence a lock is required.
+    2. The user needs to manually start an async compaction/clustering application, which means that the user needs to
+       maintain two jobs.
+    3. With the increase in the number of HUDI jobs, there is no unified service to manage compaction/clustering jobs (
+       monitor, retry, history, etc...), which will make maintenance costs increase.
+
+With this effort, we want to provide an independent compaction/clustering Service, it will have these abilities:
+
+- Provides a pluggable execution interface that can adapt to multiple execution engines, such as Spark and Flink.
+
+- With the ability to failover, need to be persisted compaction/clustering message.
+
+- Perfect metrics and reuse HoodieMetric expose to the outside.
+
+- Provide automatic failure retry for compaction/clustering job.
+
+## Implementation
+
+### Processing mode
+Different processing modes depending on whether the meta server is enabled
+
+- Enable meta server
+    - The pull-based mechanism works for fewer tables. Scanning 1000s of tables for possible services is going to induce lots of a load of listing.
+    - The meta server provides a listener that takes as input the uris of the Table Management Service and triggers a callback through the hook at each instant commit, thereby calling the Table Management Service to do the scheduling/execution for the table.
+    ![](service_with_meta_server.png)
+
+- Do not enable meta server
+    - for every write/commit on the table, the table management server is notified.
+      We can set a heartbeat timeout for each hoodie table, and if it exceeds it, we will actively pull it once to prevent the commit request from being lost
+    ![](service_without_meta_server.png)
+
+### Processing flow
+
+- After receiving the request, the table management server schedules the relevant table service to the table's timeline
+- Persist each table service into an instance table of Table Management Service
+- notify a separate execution component/thread can start executing it
+- Monitor task execution status, update table information, and retry failed table services up to the maximum number of times
+
+### Storage
+
+- There are two types of stored information
+    - Register with the hoodie table of the Table Management Service
```

Comment:
```suggestion
    - Register Hudi tables with Table Management Service
```

please use `Hudi` instead of `hoodie`. the latter is a code-level namespace.